exec/java-exec/src/test/java/org/apache/drill/TestJoinNullable.java

```diff
@@ -568,6 +570,22 @@ public void nullMixedComparatorEqualJoinHelper(final String query) throws Except
         .go();
   }
 
+  /** InnerJoin with empty dir table on nullable cols, MergeJoin */
+  // TODO: the same tests should be added for HashJoin operator, DRILL-6070
+  @Test
```

Comment:
Please add test for merge join with ignore annotation. Also please check that NLJ also works fine. If not, please fix.

Reply:
The bug was founded for NLJ and empty tables. I have resolved that issue.
The separate test class is added for empty dir tables and different join operators.

Also I have made refactoring for the TestHashJoinAdvanced, TestMergeJoinAdvanced, TestNestedLoopJoin classes.